src/multisig.ts

```diff
@@ -281,32 +281,21 @@ export function mergeMultisigTransactions(multisigTxnBlobs: Uint8Array[]) {
     }
 
     // now, we can merge
-    newSubsigs = unisig.msig.subsig.map((uniSubsig, index) => {
-      const current = refSigTx.msig.subsig[index];
-      if (current.s) {
-        // we convert the Uint8Arrays uniSubsig.s and current.s to Buffers here because (as
-        // of Dec 2020) React overrides the buffer package with an older version that does
-        // not support Uint8Arrays in the comparison function. See this thread for more
-        // info: https://github.com/algorand/js-algorand-sdk/issues/252
-        if (
-          uniSubsig.s &&
-          Buffer.compare(Buffer.from(uniSubsig.s), Buffer.from(current.s)) !== 0
-        ) {
-          // mismatch
-          throw new Error(MULTISIG_MERGE_SIG_MISMATCH_ERROR_MSG);
-        }
-        return {
-          pk: current.pk,
-          s: current.s,
-        };
+    unisig.msig.subsig.forEach((uniSubsig, index) => {
+      if (!uniSubsig.s) return;
+      const current = newSubsigs[index];
+      // we convert the Uint8Arrays uniSubsig.s and current.s to Buffers here because (as
+      // of Dec 2020) React overrides the buffer package with an older version that does
+      // not support Uint8Arrays in the comparison function. See this thread for more
+      // info: https://github.com/algorand/js-algorand-sdk/issues/252
+      if (
+        current.s &&
+        Buffer.compare(Buffer.from(uniSubsig.s), Buffer.from(current.s)) !== 0
+      ) {
+        // mismatch
+        throw new Error(MULTISIG_MERGE_SIG_MISMATCH_ERROR_MSG);
       }
-      if (uniSubsig.s) {
-        return {
-          pk: current.pk,
-          s: uniSubsig.s,
-        };
-      }
-      return current;
+      current.s = uniSubsig.s;
```

Comment:
Minor, as this doesn't affect correctness: this line always modifies `current`, but really it only needs to modify it if `current.s` is undefined.

Reply:
Same question as above, this is a O(1) operation that is sometime redundant, I could add an if statement that would run... in constant time. 
This is a little against my coding style as I try to write things the most concise way but I can make the change if you feel it adds clarity.

Reply:
I just think it's a bit odd that the above lines ensure that `uniSubsig.s` and `current.s` (if defined) are equivalent, then this line assigns them.

However I don't feel strongly about this, so if you prefer to keep it as is, that's fine.

Reply:
cool ok